spec/project-spec.coffee

```diff
@@ -197,6 +199,26 @@ describe "Project", ->
           atom.project.bufferForPath("b").then (anotherBuffer) ->
             expect(anotherBuffer).not.toBe buffer
 
+  describe ".repositoryForDirectory(directory)", ->
+    it "resolves to null when the directory does not have a repository", ->
+      waitsForPromise ->
+        directory = new Directory("/tmp")
+        atom.project.repositoryForDirectory(directory).then (result) ->
+          expect(result).toBeNull()
+          expect(atom.project.repositoryProviders.length).toBeGreaterThan 0
+          expect(atom.project.repositoryPromisesByPath.size).toBe 0
+
+    it "resolves to a GitRepository and is cached when the given directory is a Git repo", ->
+      waitsForPromise ->
+        directory = new Directory(path.join(__dirname, '..'))
+        atom.project.repositoryForDirectory(directory).then (result) ->
+          expect(result).toBeInstanceOf GitRepository
+          dirPath = directory.getRealPathSync()
+          expect(result.getPath()).toBe path.join(dirPath, '.git')
+
+          # Verify that the result is cached.
+          expect(atom.project.repositoryForDirectory(directory)).toBe(atom.project.repositoryForDirectory(directory))
```

Comment:
I added the variable back and shortened the assertion.

Made sure entire file still passes.